actors/user/src/main/java/org/sunbird/user/service/impl/FeedServiceImpl.java

```diff
@@ -1,24 +1,108 @@
 import org.sunbird.dto.SearchDTO;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.util.Util;
 import org.sunbird.models.user.Feed;
 import org.sunbird.user.service.IFeedService;
+import scala.concurrent.Future;
 
 public class FeedServiceImpl implements IFeedService {
-    @Override
-    public Response save(Feed feed) {
-        return null;
-    }
+  private ElasticSearchService esService = EsClientFactory.getInstance(JsonKey.REST);
+  private Util.DbInfo usrFeedDbInfo = Util.dbInfoMap.get(JsonKey.USER_FEED_DB);
+  private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private ObjectMapper mapper = new ObjectMapper();
 
-    @Override
-    public Response search(SearchDTO searchDTO) {
-        return null;
+  @Override
+  public Response save(Feed feed) {
+    Response response = null;
+    if (StringUtils.isNotBlank(feed.getId())) {
+      ProjectLogger.log(
+          "FeedServiceImpl:save Calling updateUserFeed for feed id : " + feed.getId(),
+          LoggerEnum.INFO.name());
+      response = updateUserFeed(feed);
+    } else {
+      ProjectLogger.log("FeedServiceImpl:save Calling insertUserFeed", LoggerEnum.INFO.name());
+      response = insertUserFeed(feed);
     }
+    return response;
+  }
+
+  private Response insertUserFeed(Feed feed) {
+    Map<String, Object> userFeed = mapper.convertValue(feed, Map.class);
+    userFeed.put(JsonKey.ID, ProjectUtil.generateUniqueId());
+    userFeed.put(JsonKey.CREATED_ON, new Timestamp(Calendar.getInstance().getTimeInMillis()));
+    Response response = saveFeed(userFeed);
+    esService.save(ProjectUtil.EsType.userfeed.getTypeName(), feed.getId(), userFeed);
+    return response;
+  }
 
-    @Override
-    public boolean delete(String id) {
-        return false;
+  private Response updateUserFeed(Feed feed) {
+    ProjectLogger.log(
+        "FeedServiceImpl:updateUserFeed fetching getRecordById for feed id " + feed.getId(),
+        LoggerEnum.INFO.name());
+    Response response =
+        cassandraOperation.getRecordById(
+            usrFeedDbInfo.getKeySpace(), usrFeedDbInfo.getTableName(), feed.getId());
+    Map<String, Object> userFeed = new HashMap<>();
+    if (null != response && null != response.getResult()) {
+      List<Map<String, Object>> userFeedList =
+          (List<Map<String, Object>>) response.getResult().get(JsonKey.RESPONSE);
+      if (CollectionUtils.isNotEmpty(userFeedList)) {
+        userFeed = userFeedList.get(0);
+      }
+    }
+    String feedData = (String) userFeed.get(JsonKey.FEED_DATA);
+    Map<String, Object> feedDataMap = new HashMap<>();
+    try {
+      feedDataMap = mapper.readValue(feedData, new TypeReference<Map<String, Object>>() {});
+    } catch (Exception ex) {
+      ProjectLogger.log(
+          "FeedServiceImpl:save: Exception occurred while converting feed data from db to List",
+          ex);
     }
+    List<String> userChannel = (List<String>) feedDataMap.get(JsonKey.CHANNEL);
```

Comment:
channel key change to prospectChannels , so make sure u are align with it.
and this is optional , so make sure , if this key not present then you don't need to append it.